refinery/core/views.py

```diff
@@ -1040,6 +1043,104 @@ def post(self, request):
         return Response(token)
 
 
+class SiteProfileViewSet(APIView):
+    """API endpoint that allows for SiteProfileViewSet to be edited.
+     ---
+    #YAML
+
+    PATCH:
+        parameters_strategy:
+        form: replace
+        query: merge
+
+        parameters:
+            - name: about_markdown
+              description: Markdown paragraph
+              type: string
+              paramType: form
+              required: false
+            - name: intro_markdown
+              description: Markdown paragraph
+              type: string
+              paramType: form
+              required: false
+            - name: twitter_username
+              description: twitter user name to display twitter feed
+              type: string
+              paramType: form
+              required: false
+            - name: site_videos
+              description: string object with source, source ids, and captions
+              type: string
+              paramType: form
+              required: false
+    ...
+    """
+    http_method_names = ["get", "patch"]
+
+    def get(self, request):
+        site_profile = SiteProfile.objects.get(site=get_current_site(request))
+        serializer = SiteProfileSerializer(site_profile)
+        return Response(serializer.data)
+
+    def patch(self, request):
+        if not request.user.is_superuser:
+            return Response(
+                self.request.user, status=status.HTTP_401_UNAUTHORIZED
+            )
+
+        try:
+            site_profile = SiteProfile.objects.get(
+                site=get_current_site(request)
+            )
+        except SiteProfile.DoesNotExist as e:
+            logger.error("Site profile for the current site does not exist.")
+            return HttpResponseNotFound(e)
+        except SiteProfile.MultipleObjectsReturned:
+            logger.critical("Multiple site profiles for current site error.")
+            return HttpResponseServerError(e)
+
+        site_videos = request.data.get('site_videos')
+        # remove unlisted videos
+        if site_videos is not None:
+            db_site_videos = SiteVideo.objects.filter(
+                site_profile=site_profile
+            )
+            new_video_list = json.loads(request.data.getlist('site_videos')[0])
+            new_video_list_ids = [vid.get('id') for vid in new_video_list]
+            # delete unused videos
+            for video in db_site_videos:
+                if video.id not in new_video_list_ids:
+                    video.delete()
```

Comment:
If I'm reading this correctly an existing video will be deleted if its `id` isn't included in a given PATCH payload? So one would have to include all existing `site_videos` in a patch to preserve them even if they aren't being updated?